### ISSUE TITLE
Intellij Ultimate + Go: Constant re-sync is necessary to keep external packages resolving

### ISSUE THREAD

**Issue by arjantop-cai** [bazelbuild/intellij#2057]
After re-sync all 3rd party dependencies are resolved fine in intellij, if I run some specific tests or targets in console they everything is highlighted in red immediately. Intellij reports and error `Build constraints exclude all Go files in '' `.

This makes the experience of using bazel in intellij pretty unusable, every time you run a command in bazel you have to re-sync. Cannot say when this broke, but it was working 6 months ago.

Intellij Ultimate 2019.3.5 (Same with 2020.1)
Latest Go Plugin
bazel 3.4.1

**Comment by humphrej** [comment]
I see something similar in GoLand 2020.1.4 - external dependencies appear in the editor as red "Unresolved dependency XXX", after I run certain Run targets (not all).  To fix I need to Bazel resync.  I don't see any error though.

**Comment by v3n** [comment]
Is this possible related to `rules_go` changing paths based on build configuration?

**Comment by dejan-lokar** [comment]
I experience the same problem with Bazel plugin 2020.07.13.0.2 and GoLand 2020.1.4.

**Comment by lucascarvalho** [comment]
I'm also experiencing the same issue

```
IntelliJ IDEA 2020.1.4 (Ultimate Edition)
Build #IU-201.8743.12
```

```
Bazel plugin: 2020.11.17.0.2
```

**Comment by glcohen** [comment]
I experience this issue multiple times a day.

GoLand 2020.2.4
Bazel plugin 2021.01.29.1.1

**Comment by MichaelSnowden** [comment]
I am also getting this issue multiple times a day:

GoLand 2020.2.4
Bazel plugin 2020.12.01.01

**Comment by fkorotkov** [comment]
I think this is the same issue as #1817. I've attempted to fix it in #3165.
